sklearn/ensemble/iforest.py

```diff
@@ -414,37 +418,56 @@ def _compute_chunked_score_samples(self, X):
         return scores
 
     def _compute_score_samples(self, X, subsample_features):
-        """Compute the score of each samples in X going through the extra trees.
+        """Compute the score of each samples in X going through the extra
+        trees.
 
         Parameters
         ----------
         X : array-like or sparse matrix
-
         subsample_features : bool,
             whether features should be subsampled
+
+        Returns
+        -------
+        ndarray
+            The anomaly scores for each sample
         """
-        n_samples = X.shape[0]
+        def get_depths(_X, trees, trees_features, _subsample_features):
+            n = _X.shape[0]
+            batch_depths = np.zeros(n, order="f")
+
+            for tree, features in zip(trees, trees_features):
+                X_subset = _X[:, features] if _subsample_features else _X
+
+                leaves_index = tree.apply(X_subset)
+                node_indicator = tree.decision_path(X_subset)
+                n_samples_leaf = tree.tree_.n_node_samples[leaves_index]
+
+                batch_depths += np.ravel(node_indicator.sum(axis=1)) \
+                    + _average_path_length(n_samples_leaf) - 1.0
+
+            return batch_depths
+
+        n_jobs, n_estimators, starts = _partition_estimators(
+            self.n_estimators, self.n_jobs)
```

Comment:
Looking at this issue again, I think we can do this:

```suggestion
            self.n_estimators, None)
```

which allows `joblib.parallel_backend` to control `n_jobs`. At a higher level, we can use `parallel_backend` to control `n_jobs`:

```python
with parallel_backend("loky", n_jobs=6):
    iso.score_samples(X)
```

**Details**: Seeing that `_partition_estimators` uses `effective_n_jobs`:

https://github.com/scikit-learn/scikit-learn/blob/8d295fbdc77c859b2ee811b2ee0588c48960bf6a/sklearn/ensemble/_base.py#L209

`effective_n_jobs` queries configuration as follows:

```python
with parallel_backend("loky", n_jobs=4):
    print(effective_n_jobs(None))
# 4

# default is 1
print(effective_n_jobs(None))
# 1
```